architecture.md

```diff
@@ -0,0 +1,78 @@
+# Ignition Gazebo Architecture
+
+Ignition Gazebo is the application entry-point of Ignition architecture, generally encompassing the use of all other Ignition libraries.
+It is a library itself, but its use case described in this document is as an executable.
+As an executable, it runs a simulation by launching two processes: the backend server process and frontend client process.
+
+The core of Ignition Gazebo is the server-client communication and the various plugins it loads to facilitate the simulation.
+Plugins are loaded at runtime to create a link between other Ignition libraries and Ignition Gazebo so they can exchange information.
+Users can write their own plugins.
+Because they’re loaded at runtime, Ignition does not need to be recompiled to add or remove plugins.
```

Comment:
Might be good to clarify that the users' plugins are compiled into library files. Right now it reads like a missing link, between the users writing the plugins, and the plugins magically getting loaded at runtime, but it's unclear how they turned from code into something loadable.

Reply:
I was wondering myself what ties the plugins to the libraries. Will definitely be adding this!